crates/pixi_config/src/lib.rs

```diff
@@ -909,10 +910,11 @@ impl Config {
     }
 
     /// Merge the given config into the current one.
+    /// The given config will have higher priority
     #[must_use]
-    pub fn merge_config(mut self, other: Config) -> Self {
-        self.mirrors.extend(other.mirrors);
-        self.loaded_from.extend(other.loaded_from);
+    pub fn merge_config(self, mut other: Config) -> Self {
```

Comment:
This just flips all logic. Please make sure the logic stays correct over all calls. I'm assuming the only change required is in `load()`